app/models/ext_management_system.rb

```diff
@@ -677,6 +687,12 @@ def destroy_queue
     msg = "Queuing destroy of #{self.class.name} with id: #{id}"
 
     _log.info(msg)
+
+    # Before we queue the `#destroy` pause the provider to prevent a provider in
+    # a bad state from filling up the queue preventing the `#destroy` from being
+    # processed.
+    pause_queue!(:priority => MiqQueue::HIGH_PRIORITY)
```

Comment:
I think we have to also change the destroy methods itself to verify that the provider is paused before trying to destroy it.  Otherwise if the pause queue item and the destroy queue item are run simultaneously, they may race each other.  I think the destroy method should just put itself back on the queue if it's not paused.  Wdyt?

Reply:
Yeah true, if they ran at the same time the pause could "resurrect" the EMS after it was destroyed like we had with the workers.

Reply:
We are now re-queueing the destroy if the pause hasn't run yet: https://github.com/ManageIQ/manageiq/pull/21276/files#diff-8d286308b75ae7dc8a266753732b0c1709d33293518c479765e34cadf8207910R731
Spec for this case: https://github.com/ManageIQ/manageiq/pull/21276/files#diff-f50b4b351bf43b590f4afa70fdf2534bb4291339af5d57f9c283b73913a95ab0R777-R794